symfit/core/fit.py

```diff
@@ -105,10 +102,118 @@ def __init__(self, model, constraint_type=None):
             # should be introduced to fulfill the same role.
             model = {Variable(): expr for expr in model}
         self._init_from_dict(model)
-        if constraint_type is not None and len(self.dependent_vars) != 1:
+
+    @classmethod
+    def as_constraint(cls, constraint, model, constraint_type=None, **init_kwargs):
+        """
+        Initiate a Model which should serve as a constraint. Such a
+        constraint-model should be initiated with knowledge of another
+        ``BaseModel``, from which it will take its parameters::
+
+            model = Model({y: a * x + b})
+            constraint = Model.as_constraint(Eq(a, 1), model)
+
+        ``constraint.params`` will be ``[a, b]`` instead of ``[a]``.
+
+        :param constraint: An ``Expr``, a mapping or iterable of ``Expr``, or a
+            ``Relational``.
+        :param model: An instance of (a subclass of)
+            :class:`~symfit.core.fit.BaseModel`.
+        :param constraint_type: Optional: as what kind of constraint should this
+            model be interpreted when used as such? Required when ``constraint``
+            is not a :class:`~sympy.core.relational.Relational`.
+        :param kwargs: Any additional keyword arguments which will be passed on
+            to the init method.
+        """
+        allowed_types = [sympy.Eq, sympy.Ge, sympy.Le]
+
+        if isinstance(constraint, Relational):
+            constraint_type = constraint.__class__
+            constraint = constraint.lhs - constraint.rhs
+        elif constraint_type is None:
+            raise TypeError('Please provide a ``constraint_type``.')
+
+        # Initiate the constraint model, in such a way that we take care
+        # of any dependencies
+        self = cls.with_dependencies(constraint,
+                                     dependency_model=model,
+                                     **init_kwargs)
+
+        # Check if the constraint_type is allowed, and flip the sign if needed
+        if constraint_type not in allowed_types:
+            raise ModelError(
+                'Only constraints of the type {} are allowed. A constraint'
+                ' of type {} was provided.'.format(allowed_types,
+                                                   constraint_type)
+            )
+        elif constraint_type is sympy.Le:
+            # We change this to a Ge and flip the sign
+            self = - self
```

Comment:
Good remark, not anymore because the minus is applied to the whole model_dict, but from this PR onwards it should only apply to the dependent component of the model_dict. I'll update `__neg__`

Reply:
My problem is more conceptual:
```python3
class A:
    def __init__(self):
        self = 2

a = A()
print(a)
```
What is `a`?
Maybe doing `self *= -1` is more sane, since that is explicitely in-place (and possibly `__ineg__` instead of `__neg__`.

Reply:
Ah, you mean that I'm messing with `self`? I see the confusion, but note that this is a classmethod right? So this is an alternative constructor which has to return an instance in the end. I already call that `self` because that is what it will end up becoming, but we can still manipulate it however we want at this stage.

Reply:
And I we should keep supporting pure negation of models I think, because if you want to maximize something you can just say `Fit(- model, ...)` which is rather nice.

Reply:
Aaaah. I missed it's a class method. In that case change all `self` to `obj`. And yes, negation of models is an important feature.